### PR TITLE
setup namespace, rbac and roles for the tekton chains

### PR DESCRIPTION
**Pull request by harshad16**
setup namespace, rbac and roles for the tekton chains
Signed-off-by: Harshad Reddy Nalla <hnalla@redhat.com>

Related-to: https://github.com/operate-first/support/issues/438

Tekton chains provide security to our pipeline supply chains.
It would sign the images based on the secrets. 
https://github.com/tektoncd/chains#installation

### CODE REVIEW

**Comment by harshad16** [comment]
cc: @screeley44
This would provide permission to tekton chains to look up the pipelines running on the cluster.

File: cluster-scope/base/core/namespaces/tekton-chains/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+---
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+namespace: tekton-chains
+
+components:
```
Comment (HumairAK) [inline cluster-scope/base/core/namespaces/tekton-chains/kustomization.yaml:7]: can we add a quota to this?

Comment (harshad16) [reply cluster-scope/base/core/namespaces/tekton-chains/kustomization.yaml:7]: yes of course :)

File: cluster-scope/overlays/prod/moc/smaug/kustomization.yaml [hunk 2]
```diff
@@ -105,6 +105,7 @@ resources:
   - ../../../../base/rbac.authorization.k8s.io/clusterrolebindings/moc-nfs-democratic-csi-node-rb
   - ../../../../base/rbac.authorization.k8s.io/clusterrolebindings/node-labeler
   - ../../../../base/rbac.authorization.k8s.io/clusterrolebindings/scc-privileged-opf-ci-pipelines-aicoe-ci
+  - ../../../../base/rbac.authorization.k8s.io/clusterrolebindings/tekton-chains
```
[the remark this answers is not in the document]
Comment (harshad16) [reply cluster-scope/overlays/prod/moc/smaug/kustomization.yaml:113]: Thanks for the catch

**Comment by harshad16** [comment]
Ready for the Re-Review

**Pull request merged**